components/accelerator/nvidia/query/detect.go

```diff
@@ -0,0 +1,145 @@
+package query
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/leptonai/gpud/log"
+	"github.com/leptonai/gpud/pkg/file"
+	"github.com/leptonai/gpud/pkg/process"
+
+	"github.com/NVIDIA/go-nvlib/pkg/nvlib/device"
+	nvinfo "github.com/NVIDIA/go-nvlib/pkg/nvlib/info"
+	"github.com/NVIDIA/go-nvml/pkg/nvml"
+)
+
+// Returns true if the local machine has NVIDIA GPUs installed.
+func GPUsInstalled(ctx context.Context) (bool, error) {
+	smiInstalled := SMIExists()
+	if !smiInstalled {
+		return false, nil
+	}
+	log.Logger.Info("nvidia-smi installed")
+
+	// now that nvidia-smi installed,
+	// check the NVIDIA GPU presence via PCI bus
+	pciDevices, err := ListPCIs(ctx)
+	if err != nil {
+		return false, err
+	}
+	if len(pciDevices) == 0 {
+		return false, nil
+	}
+	log.Logger.Infow("nvidia PCI devices found", "devices", len(pciDevices))
+
+	// now that we have the NVIDIA PCI devices,
+	// call NVML C-based API for NVML API
+	productName, err := LoadProductName(ctx)
+	if err != nil {
+		return false, err
+	}
+	log.Logger.Infow("detected nvidia gpu", "product", productName)
```

Comment:
productName can also be a network card. do we check the type of the device?

Reply:
It's calling the device.GetName, so changed this logging to GPU device name.